src/mono/wasm/runtime/lazyLoading.ts

```diff
@@ -0,0 +1,54 @@
+/* eslint-disable no-prototype-builtins */
+
+import { INTERNAL, loaderHelpers, runtimeHelpers } from "./globals";
+import type { WebAssemblyResourceLoader } from "./loader/blazor/WebAssemblyResourceLoader";
+
+export async function loadLazyAssembly(assemblyNameToLoad: string): Promise<boolean> {
+    const resourceLoader: WebAssemblyResourceLoader = INTERNAL.resourceLoader;
+    const resources = resourceLoader.bootConfig.resources;
+    const lazyAssemblies = resources.lazyAssembly;
+    if (!lazyAssemblies) {
+        throw new Error("No assemblies have been marked as lazy-loadable. Use the 'BlazorWebAssemblyLazyLoad' item group in your project file to enable lazy loading an assembly.");
+    }
+
+    const assemblyMarkedAsLazy = lazyAssemblies.hasOwnProperty(assemblyNameToLoad);
+    if (!assemblyMarkedAsLazy) {
+        throw new Error(`${assemblyNameToLoad} must be marked with 'BlazorWebAssemblyLazyLoad' item group in your project file to allow lazy-loading.`);
+    }
+
+    if (loaderHelpers.loadedAssemblies.some(f => f.includes(assemblyNameToLoad))) {
+        return false;
```

Comment:
I think you just want to filter it out, not quit for all the others.

Reply:
This function loads a singe assembly

Reply:
oh, `loadedAssemblies` is `O(n*n)` please replace it with `Map`

Reply:
It's not `O(n*n)`. The `loaderHelpers.loadedAssemblies` is `string[]` and I'm searching for substring in each string

Reply:
`loadLazyAssembly` is called `M` times, `loadedAssemblies` has `N` members, each is then `P` characters long.

Reply:
I also wonder about assembly name with extensions `.dll` and `.wasm` and also without extension. Would it all work ?

Reply:
this is probably small fish, feel free to ignore it if the solution is too complex.

Reply:
I'm going to ignore it for the moment. We are talking about array of tens of items.